ffmpeg/transcoder.c

```diff
@@ -168,15 +168,21 @@ int transcode(struct transcode_thread *h,
 
       // first segment of a stream, need to initalize output HW context
       // XXX valgrind this line up
-      if (!h->initialized || AV_HWDEVICE_TYPE_NONE == octx->hw_type) {
+      if (!h->initialized || AV_HWDEVICE_TYPE_NONE == octx->hw_type && !ictx->transmuxing) {
         ret = open_output(octx, ictx);
         if (ret < 0) LPMS_ERR(transcode_cleanup, "Unable to open output");
+        if (ictx->transmuxing) {
+          octx->oc->flags |= AVFMT_FLAG_FLUSH_PACKETS;
```

Comment:
Trying to reduce buffering inside ffmpeg - we're using transmuxing to stream from internet and back out to internet - so no point to buffer data.
Probably doesn't matter much, just in case.